.github/workflows/ci.yml

```diff
@@ -0,0 +1,125 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - '[0-9]+.x'
+      - '[0-9]+.[0-9]+.x'
+  pull_request:
+    types: [opened, reopened, synchronize]
+    branches:
+      - master
+      - '[0-9]+.x'
+      - '[0-9]+.[0-9]+.x'
```

Comment:
I'm not sure how much of the above `on` stuff we actually need... maybe it's fine to just run this on all branches and just limit the jobs down below to only run under more specific circumstances -- for example only deploy from `master` or `a.b.c` release branches.

Reply:
Yeah I was wondering if it'd be nice to divide into two workflows - one that runs when a PR is opened and added to, and then one were `master` or a version branch is committed to (i.e. merge).  Since we usually run a different set of tests for PR CI versus merge.

Reply:
I like that... no need to lint on `master` since in theory nothing should have been able to merge into it without passing the stricter CI. Want me to split this out into two now or do you want to just add to this PR as you make progress with your stuff?

Reply:
Yeah no rush, I can split it out once I get deploy working... 🤞

Reply:
You can add `if:` to steps and check the branch name. [Here's an example of that](https://github.com/Brightspace/d2l.dev/blob/master/.github/workflows/paste.d2l.dev.yml#L31).

Sometimes its nice to have one big workflow but disable some steps based on the scenario... but it can get overboard and one workflow might be right.

Reply:
P.S. for pull requests this is filtering on the source branch, not the target branch.... so you probably don't want `branches:`